src/version.rs

```diff
@@ -106,21 +106,94 @@ impl<'de> Deserialize<'de> for Identifier {
     }
 }
 
+/// Pre-release or build metadata, consisting of a collection of `Identifiers`.
+#[derive(Clone, PartialEq, Eq, Debug, Hash, PartialOrd, Ord)]
+pub struct MetaIdentifier(Vec<Identifier>);
+
+type Iter<'a> = ::std::slice::Iter<'a, Identifier>;
```

Comment:
It's a little more boilerplate, but I think we should make these opaque new-types like `Iter<'a>(::std::slice::Iter<'a, Identifier>)` so that we don't leak the fact that the internal storage is a `Vec`.